infrastructure/src/test/java/org/corfudb/infrastructure/CorfuSequencerServerTest.java

```diff
@@ -0,0 +1,528 @@
+        assertThat(response.getPayload().getBootstrapSequencerResponse().getRespType())
+                .isEqualTo(Sequencer.BootstrapSequencerResponseMsg.Type.NACK);
+    }
+
+    /**
+     * Test that the SequencerServer correctly handles a BootstrapSequencerRequestMsg. If the
+     * epoch sent by the client is greater than or equal to the current server epoch, and bootstrapWithoutTailsUpdate
+     * is set as false, a BootstrapSequencerResponseMsg.ACK message should be received.
+     * <p>
+     * Further, we request the server with a StreamsAddressRequestMsg and verify that
+     * the globalLogTail, trimMark and the streamsAddressMap were correctly updated on the server side.
+     */
+    @Test
+    public void testBootStrapSequencer() {
+        when(spyFactoryHelper.getStreamAddressSpaceMap()).thenReturn(mockStreamAddressSpaceMap);
+        sequencerServer = new SequencerServer(mockServerContext, spyFactoryHelper);
+        UUID streamA = UUID.nameUUIDFromBytes("streamA".getBytes());
+        // Construct new tails
+        Map<UUID, StreamAddressSpace> tailMap = new HashMap<>();
+
+        long globalTail = 0;
+        long bootstrapMsgEpoch = 1;
+        StreamAddressSpace streamAddressSpace = new StreamAddressSpace(Address.NON_ADDRESS,
+                Roaring64NavigableMap.bitmapOf(globalTail));
+        tailMap.put(streamA, streamAddressSpace);
+
+        RequestMsg request = getRequestMsg(
+                getBasicHeader(false, true),
+                CorfuProtocolSequencer.getBootstrapSequencerRequestMsg(
+                        tailMap,
+                        globalTail,
+                        bootstrapMsgEpoch,
+                        false)
+        );
+
+        ArgumentCaptor<ResponseMsg> responseCaptor = ArgumentCaptor.forClass(ResponseMsg.class);
+        sequencerServer.handleMessage(request, mockChannelHandlerContext, mockServerRouter);
+
+        // Verify that sendResponse() was called and capture the response object
+        verify(mockServerRouter)
+                .sendResponse(responseCaptor.capture(), any(ChannelHandlerContext.class));
+        ResponseMsg response = responseCaptor.getValue();
+        // Assert that the payload has a BOOTSTRAP_SEQUENCER_RESPONSE and that the base
+        // header fields have remained the same
+        assertThat(compareBaseHeaderFields(request.getHeader(), response.getHeader())).isTrue();
+        assertThat(response.getPayload().hasBootstrapSequencerResponse()).isTrue();
+        assertThat(response.getPayload().getBootstrapSequencerResponse().getRespType())
+                .isEqualTo(Sequencer.BootstrapSequencerResponseMsg.Type.ACK);
+
+        // Verify that the mockServerContext.setSequencerEpoch() was called and inspect the argument
+        ArgumentCaptor<Long> longArgumentCaptor = ArgumentCaptor.forClass(Long.class);
+        verify(mockServerContext).setSequencerEpoch(longArgumentCaptor.capture());
```

Comment:
I don't think this argument captor is needed. See similar comments.

Reply:
Done, Thank you.